impeller/renderer/texture.h

```diff
@@ -20,7 +20,10 @@ class Texture {
   virtual void SetLabel(const std::string_view& label) = 0;
 
   [[nodiscard]] virtual bool SetContents(const uint8_t* contents,
```

Comment:
I think the subclass should verify the slice is appropriate for the descriptor before letting the subclasses set the contents for that slice. Perhaps add a new pure virtual `OnSetContents` and make this non-virtual?

Reply:
Agreed and done.